src/lib/caching/structures/guilds/GuildMember.ts

```diff
@@ -85,6 +85,73 @@ export class GuildMember extends Structure {
 		return this.client.users.get(this.id) ?? null;
 	}
 
+	/**
+	 * The calculated permissions from the member's {@link GuildMemberRoleStore roles}.
+	 * @since 0.0.1
+	 */
+	public get permissions(): Permissions {
+		if (this.id === this.guild.ownerID) return new Permissions(Permissions.ALL).freeze();
+		return new Permissions(this.roles.map(role => role.permissions));
```

Comment:
Should return frozen permissions.

Reply:
This also is (the only other place) impacted by checkAdmin. Instead of a checkAdmin option on has, which is really a footgun, we should check if roles grant administrator here, and return Permissions.ALL (frozen) here.

Reply:
Support to remove the Permissions#has overwrite in this pr if you would like.

Reply:
Is the change 9482a5b correct? I don't really know what the issues are.